legal-api/src/legal_api/services/filings/validations/change_of_directors.py

```diff
@@ -70,43 +71,47 @@ def validate_directors_addresses(cod: Dict) -> List:
 def validate_effective_date(business: Business, cod: Dict) -> List:
     """Return an error or warning message based on the effective date validation rules.
 
-    Rules:
+    Rules: (text from the BA rules document)
         - The effective date of change cannot be in the future.
-        - The effective date cannot be a date prior to their Incorporation Date
+        - The effective date cannot be a date prior to their Incorporation Date.
         - The effective date of change cannot be a date that is farther in the past
-            as a previous COD filing (Standalone or AR).
+            than a previous COD filing (standalone or AR).
         - The effective date can be the same effective date as another COD filing
-            (standalone OR AR). If this is the case:
-        - COD filing that was filed most recently as the most current director information.
+            (standalone or AR). If this is the case:
+        - COD filing that was filed most recently is the most current director information.
     """
+    msg = []
+
+    # get effective datetime string from filing
     try:
-        filing_effective_date = cod['filing']['header']['effectiveDate']
+        effective_datetime_str = cod['filing']['header']['effectiveDate']
     except KeyError:
-        try:
-            # we'll assume the filing is at 0 hours UTC
-            filing_effective_date = cod['filing']['header']['date'] + 'T00:00:00+00:00'
-        except KeyError:
-            return {'error': babel('No effective_date or filing date provided.')}
+        return {'error': babel('No effective date provided.')}
 
+    # convert string to datetime
     try:
-        effective_date = datetime.fromisoformat(filing_effective_date)
+        effective_datetime_utc = datetime.fromisoformat(effective_datetime_str)
     except ValueError:
-        return {'error': babel('Invalid ISO format for effective_date or filing date.')}
+        return {'error': babel('Invalid ISO format for effective date.')}
 
-    msg = []
-
-    # The effective date of change cannot be in the future
-    if effective_date > datetime.utcnow():
+    # check if effective datetime is in the future
+    if effective_datetime_utc > datetime.utcnow():
         msg.append({'error': babel('Filing cannot have a future effective date.')})
 
-    # The effective date cannot be a date prior to their Incorporation Date
-    if effective_date < business.founding_date:
-        msg.append({'error': babel('Filing cannot be before a businesses founding date.')})
+    # convert to legislation timezone and then get date only
+    effective_date_leg = LegislationDatetime.as_legislation_timezone(effective_datetime_utc).date()
+
+    # check if effective date is before their incorporation date
+    founding_date_leg = LegislationDatetime.as_legislation_timezone(business.founding_date).date()
+    if effective_date_leg < founding_date_leg:
```

Comment:
I don't even know how we'd deal with this... the COD doesn't have a time, so how would you mark that it came after an IA? Or after another COD on the same day? Probably db order (index)?
What do you suggest to do here?

Reply:
the COD should have a submission datetime.

Reply:
Yes, the COD could have an effective datetime < incorporation datetime, but isn't it up to other code to sort the transactions in order (eg, by filing id)?

I remember we once talked about injecting the "now" time into a COD's effective date but I'm not sure that will always work.

Reply:
Example:
1. I incorporated a company 1 hour ago (effective 2020-08-19 17:30 UTC).
2. I just filed a COD (effective 2020-08-19 07:00 UTC).

Is the COD valid? Yes.
What code cares about the COD effective datetime?

Reply:
As in another conversation above, I am adding a check that COD submission datetime is >= business founding datetime.

As for the COD effective date-time, yes, it can appear to be earlier the IA. This is because the COD is effective on the selected date at 00:00 local. We just have to be careful to process the COD data (esp with an IA or other CODs on the same day) in submission datetime order (or filing id order).